tardis/tests/tests_slow/report.py

```diff
@@ -0,0 +1,174 @@
+"""
+A helper class which works as a plugin to generate the test report and upload it
+to the group server's dokuwiki. It inheirts from the class `HTMLReport` of
+the `pytest-html` plugin. The test report contains the following details:
+
+* The git commit hash on which test run was executed.
+* The time of generation of test report.
+* Number of passes, fails, errors, skips etc.
+* Tabular representation of each method - name, result, duration.
+* Embedded image of plot(s) and error log below a particular method (if any).
+
+As a subclass, this class serves as a plugin and hence, `pytest-html` has to be
+unregistered during the test run for tis plugin to function.
+
+When the integration tests are selected for a particular test run, this class
+is registered as a plugin in `pytest_configure` and subsequently unregistered in
+`pytest_unconfigure`. As a plugin, it implements several "hook" functions
+specified in pytest's official documentation.
+
+
+References
+==========
+1. "Writing Plugins" ( https://pytest.org/latest/writing_plugins.html )
+2. "Hookspec Source" ( https://pytest.org/latest/_modules/_pytest/hookspec.html )
+3. "pytest-html" ( https://www.github.com/davehunt/pytest-html )
+"""
+import datetime
+import pkg_resources
+import os
+import time
+
+# For specifying error while exception handling
+from socket import gaierror
+
+from py.xml import html, raw
+from pytest_html.plugin import HTMLReport
+import tardis
+
+try:
+    import dokuwiki
+except ImportError:
+    dokuwiki = None
```

Comment:
I think that makes for convoluted code, where it is not clear why it is excepting. But I withdraw my comment for now.